extensions/azurecore/src/account-provider/auths/httpClient.ts

```diff
@@ -151,7 +152,7 @@ const networkRequestViaProxy = <T>(
 	// compose a request string for the socket
 	let postRequestStringContent: string = '';
 	if (httpMethod === HttpMethod.POST || httpMethod === HttpMethod.PUT) {
-		const body = options?.body || '';
+		const body = (new TextEncoder()).encode(JSON.stringify(options?.body || ''));
 		postRequestStringContent =
 			'Content-Type: application/x-www-form-urlencoded\r\n' +
```

Comment:
I see this is `x-www-form-urlencoded`, whereas for PUT/POST request you're passing headers as 'application/json': https://github.com/microsoft/azuredatastudio/blob/058856008b0bee1dea5227f9d2d1cda7b85bf6e3/extensions/azurecore/src/azureResource/utils.ts#L394

Shouldn't they be same?

Reply:
will investigate this

Reply:
This Content-Type definitely doesn't seem correct, that content type is supposed to be used specifically for form data and the body is supposed to be in the form `key=value&key2=value2...` (like the query string)

https://www.w3.org/TR/html401/interact/forms.html#h-17.13.4.1

Did you copy this from somewhere or was there a specific reason you chose it? If you don't have a reason for using it I definitely think you should update it (after checking it still works of course) to avoid any potential issues coming up later on.